frame/collective/src/lib.rs

```diff
@@ -710,6 +714,11 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	) -> Result<(u32, DispatchResultWithPostInfo), DispatchError> {
 		let proposal_len = proposal.encoded_size();
 		ensure!(proposal_len <= length_bound as usize, Error::<T, I>::WrongProposalLength);
+		let proposal_weight = proposal.get_dispatch_info().weight;
+		ensure!(
+			proposal_weight.all_lte(T::MaxProposalWeight::get()),
+			Error::<T, I>::WrongProposalWeight
+		);
```

Comment:
The proposal weight is already added to the weight of `propose` if `threshold < 2`. So, we would not even add this tx to a block when the weight is too big. TLDR: This should not be required.

Reply:
But yeah, to have the properties as `do_propose_proposed` it makes sense.